### PR TITLE
Scheduler api

### PR DESCRIPTION
**Pull request by velddev**
- Adds initial new scheduling SDK
- Adds feature flagging base

### CODE REVIEW

**Comment by velddev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 14
           

Complexity increasing per file
==============================
- src/Miki/Services/Scheduling/Workers/ScheduleWorker.cs  1
- src/Miki/Services/Scheduling/Models/TaskKey.cs  1
- src/Miki/Extensions/ContextExtensions.cs  2
- src/Miki/Services/Scheduling/SchedulerService.cs  7
- test/Miki.Tests/Services/SchedulerServiceTests.cs  1
- src/Miki/Services/Scheduling/Workers/ScheduleWorkerGroup.cs  2
- src/Miki/Exceptions/CommandDisabledException.cs  1
- src/Miki/Extensions/TimeExtensions.cs  1
- src/Miki/Modules/Fun/Models/Reminder.cs  1
- src/Miki/Services/Scheduling/Models/TaskPayload.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/velddev/Miki/pullRequest?prid=5263604&bid=17244076)

File: src/Miki/AppProps.cs [hunk 1]
```diff
@@ -14,8 +14,8 @@ public static class AppProps
 
         public static class Emoji
         {
-            public static string Mekos => "<:mekos:421972155484471296>";
-
+            public static string Mekos
```
Comment (Drummss) [inline src/Miki/AppProps.cs:17]: These can be const?

File: src/Miki/Modules/Fun/FunModule.cs [hunk 2]
```diff
@@ -279,8 +280,6 @@ public async Task DogAsync(IContext e)
 		[Command("gif")]
 		public async Task ImgurGifAsync(IContext e)
 		{
-            var locale = e.GetLocale();
-
             string title = e.GetArgumentPack().Pack.TakeAll();
```
Comment (Drummss) [inline src/Miki/Modules/Fun/FunModule.cs:283]: NIT; var

File: src/Miki/Modules/Fun/FunModule.cs [hunk 3]
```diff
@@ -513,13 +521,13 @@ await e.ErrorEmbed(locale.GetString("error_no_reminders"))
 
                 foreach(var x in instances)
                 {
-                    string tx = x.Context;
-                    if(x.Context.Length > 30)
+                    string tx = JsonConvert.DeserializeObject<Reminder>(x.PayloadJson).Context;
```
Comment (Drummss) [inline src/Miki/Modules/Fun/FunModule.cs]: NIT; var

File: src/Miki/Modules/Fun/FunModule.cs [hunk 4]
```diff
@@ -586,14 +604,12 @@ private async Task HelpReminderAsync(IContext e)
 
             private async Task PlaceReminderAsync(IContext e, string args)
             {
-                int splitIndex = args.ToLower().LastIndexOf(" in ");
+                int splitIndex = args.ToLower().LastIndexOf(" in ", StringComparison.Ordinal);
```
Comment (Drummss) [inline src/Miki/Modules/Fun/FunModule.cs]: NIT; var

File: src/Miki/Modules/Fun/FunModule.cs [hunk 5]
```diff
@@ -603,21 +619,21 @@ await e.ErrorEmbed(e.GetLocale().GetString("error_argument_null", "time"))
...
+							{
+                                Context = reminderText,
+                                UserId = e.GetAuthor().Id
+                            }),
+                        e.GetAuthor().Id.ToString());
 
                     await new EmbedBuilder()
                         .SetTitle($"👌 {e.GetLocale().GetString("term_ok")}")
-                        .SetDescription($"I'll remind you to **{reminderText}** in **{timeUntilReminder.ToTimeString(e.GetLocale())}**\nYour reminder code is `{id}`")
+                        .SetDescription($"I'll remind you to **{reminderText}** in **{timeUntilReminder.ToTimeString(e.GetLocale())}**\n")
```
Comment (Drummss) [inline src/Miki/Modules/Fun/FunModule.cs]: Add string to resources file.

**Review by Drummss — APPROVED** [review]
Looks good Veld, approved 👍

File: src/Miki/AppProps.cs (hunk 1 above)
Comment (velddev) [reply src/Miki/AppProps.cs:17]: That is definitely a possibility, but that falls out of the scope of the current commit.

**Comment by velddev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 2
           

Complexity increasing per file
==============================
- src/Miki/Services/Scheduling/Models/TaskPayload.cs  1
- test/Miki.Tests/Services/SchedulerServiceTests.cs  1
- src/Miki/Extensions/ContextExtensions.cs  2
- src/Miki/Services/Scheduling/Workers/ScheduleWorker.cs  1
- src/Miki/Modules/Fun/Models/Reminder.cs  1
- src/Miki/Services/Scheduling/SchedulerService.cs  7
- src/Miki/Exceptions/CommandDisabledException.cs  1
- src/Miki/Extensions/TimeExtensions.cs  2
- src/Miki/Services/Scheduling/Workers/ScheduleWorkerGroup.cs  2
- src/Miki/Services/Scheduling/Models/TaskKey.cs  1
         

Clones added
============
- test/Miki.Tests/Services/SchedulerServiceTests.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/velddev/Miki/pullRequest?prid=5263604&bid=17244076)

**Comment by velddev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 2
           

Complexity increasing per file
==============================
- src/Miki/Extensions/TimeExtensions.cs  2
- src/Miki/Modules/Fun/Models/Reminder.cs  1
- src/Miki/Services/Scheduling/Models/TaskKey.cs  1
- test/Miki.Tests/Services/SchedulerServiceTests.cs  1
- src/Miki/Services/Scheduling/Models/TaskPayload.cs  1
- src/Miki/Services/Scheduling/Workers/ScheduleWorker.cs  1
- src/Miki/Exceptions/CommandDisabledException.cs  1
- src/Miki/Services/Scheduling/Workers/ScheduleWorkerGroup.cs  2
- src/Miki/Services/Scheduling/SchedulerService.cs  7
- src/Miki/Extensions/ContextExtensions.cs  2
         

Clones added
============
- test/Miki.Tests/Services/SchedulerServiceTests.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/velddev/Miki/pullRequest?prid=5263604&bid=17244076)

**Pull request merged**